docs/faq/add_op_in_backend.md

```diff
@@ -543,53 +545,81 @@ If you use python, when you type `import mxnet as mx`, two python
 functions for invoking your backend implementation are
 generated on the fly: one is for imperative programming
 registered as `mxnet.ndarray.quadratic` or `mx.nd.quadratic` for short;
-the other one is for symbolic
-programming registered under module `mxnet.symbol.quadratic`
-or `mx.sym.quadratic` for short.
+the other one is for symbolic programming registered under
+module `mxnet.symbol.quadratic` or `mx.sym.quadratic` for short.
 
 In order to unit test it in frontend, we need to add the following code
-to the python file `test_operator.py`. Note that while testing the
-forward pass is straightforward using `mx.nd.quadratic`, testing
-the backward involves a bit of more efforts. We create a
-`quadratic` symbol and feed it into the utility function `check_numeric_gradient`.
-The utility function will perform a perturbation on the input
-and calculate the response rate of the output using the
-[finite difference method](https://en.wikipedia.org/wiki/Finite_difference_method).
-Then it will compare the gradient from the backward pass with
-the values from the finite difference method. The test
-will be successful once the comparison satisfies user specified
-relative and absolute thresholds.
+to the python file `test_operator.py`. A typical operator implementation
+tests for both the `symbol` API and the `ndarray` API. In this tutorial 
+we will go over the symbol API test. Note that while testing
+the `ndarray` API is pretty straight forward as we are testing only the
+forward pass of the operator, testing for the `symbol` API is more involved
+as it tests the complete functionality of the operator - the forward pass,
+the backward pass and the gradient computation in the operator implementation.
+To facilitate the testing of these functionalities we use three helper
+functions available in the `mxnet.test_utils` module:
+ - `check_symbolic_forward `
+ - `check_symbolic_backward`
+ - `check_numeric_gradient`
+
```

Comment:
To be accurate here, you can also perform backward pass on NDArrays via autograd, so you can actually test backward pass too. I think simply mentioning that we're performing test on the Symbol APIs with the provided helper functions is enough.

Reply:
are there operator tests that use autograd and test backward pass on ndarrays? can you please point me to it?

Reply:
https://github.com/apache/incubator-mxnet/pull/11983/files#diff-cb652780258e73a9cd08568f38929aa2R3813

Reply:
But we do not usually do it unless needed, since the numeric outputs are the same from both.